lib/auth/permissions.go

```diff
@@ -148,23 +147,24 @@ Loop:
 
 // UseSearchAsRoles extends the roles of the Checker on the current Context with
 // the set of roles the user is allowed to search as.
-func (c *Context) UseSearchAsRoles(access services.RoleGetter) error {
+func (c *Context) UseSearchAsRoles(access services.RoleGetter, clusterName string) error {
+	if len(c.Checker.GetAllowedResourceIDs()) > 0 {
+		return trace.AccessDenied("cannot search with elevated roles while already logged in with a search-based access request")
```

Comment:
It took me a second to understand what "elevated roles" means in this context. Is this to prevent a situation where I create a search-based access request, the request is granted, I log in with new roles and then try to create another search-based access request?

Reply:
Yeah, stacking search-based access requests just doesn't really make sense. In fact, you wouldn't even be able to search for new resources anyway, because all actions including List are limited to only the resources you were explicitly approved for while you are logged in with a search-based request.